src/vs/workbench/browser/positronModalDialogs/copyAsCodeModalDialog.tsx

```diff
@@ -0,0 +1,164 @@
+	commandService: ICommandService;
+	dataExplorerClientInstance: IPositronDataExplorerInstance
+	keybindingService: IKeybindingService;
+	layoutService: IWorkbenchLayoutService;
+	renderer: PositronModalReactRenderer;
+}
+
+
+/**
+ * NewCopyAsCodeModalDialog component.
+ * @param props The component properties.
+ * @returns The rendered component.
+ */
+export const NewCopyAsCodeModalDialog = (props: CopyAsCodeDialogProps) => {
+	// State hooks.
+	const instance = props.dataExplorerClientInstance.dataExplorerClientInstance;
+	const codeSyntaxOptions = instance.cachedBackendState?.supported_features?.code_syntaxes ?? [];
+
+	const [selectedSyntax, setSelectedSyntax] = useState<string>(instance.guessedSyntax.code_syntax ?? 'Select Code Syntax');
+
+	const [codeString, setCodeString] = useState<string | undefined>(undefined);
+
+	useEffect(() => {
+		const getCodeString = async () => {
+			if (!selectedSyntax) {
+				return;
+			}
+			// Execute the command to get the code string based on the selected syntax.
+			const codeString = await props.commandService.executeCommand<string>(PositronDataExplorerCommandId.CopyAsCodeAction, selectedSyntax);
+			setCodeString(codeString);
+		}
+		getCodeString();
+	}, [props.commandService, selectedSyntax]);
+
+	// Construct the syntax options dropdown entries
+	const syntaxDropdownEntries = () => {
+		if (codeSyntaxOptions.length === 0) {
+			return [];
+		}
+		return syntaxInfoToDropDownItems(codeSyntaxOptions);
+	};
+
+	const syntaxInfoToDropDownItems = (
+		syntaxes: string[]
+	): DropDownListBoxItem<string, string>[] => {
+		return syntaxes.map(
+			(syntax) =>
+				new DropDownListBoxItem<string, string>({
+					identifier: syntax,
+					value: syntax,
+				})
+		);
+	};
+
+	// Construct the syntax dropdown title.
+	const syntaxDropdownTitle = () => {
+		return selectedSyntax;
+	};
+
+	const onSelectionChanged = async (item: DropDownListBoxItem<unknown, unknown>) => {
+		const typedItem = item as DropDownListBoxItem<string, string>;
+		setSelectedSyntax(typedItem.options.identifier);
+
+		// Execute the command to get the code string based on the selected syntax.
+		const exc = await props.commandService.executeCommand<string>(PositronDataExplorerCommandId.CopyAsCodeAction, typedItem.options.identifier)
+		setCodeString(exc);
+	};
+
+	// Render.
+	return (
+		<OKCancelModalDialog
+			catchErrors
+			height={300}
+			renderer={props.renderer}
+			title={(() => localize(
+				'positronCopyAsCodeModalDialogTitle',
+				"Copy as Code"
+			))()}
+			width={400}
+			onAccept={async () => {
+				props.renderer.dispose();
+			}}
+			onCancel={() => props.renderer.dispose()}
+		>
+			<VerticalStack>
+				<DropDownListBox
+					createItem={(item) => (
+						<DropdownEntry
+							title={item.options.identifier}
+						/>
+					)}
+					entries={syntaxDropdownEntries()}
+					keybindingService={props.keybindingService}
+					layoutService={props.layoutService}
+					title={syntaxDropdownTitle()}
```

Comment:
I don't think this needs to be a function since you are always returning the same state variable: `selectedSyntax `

I think this should be fine:
```suggestion
					title={selectedSyntax}
```